WordPressKitTests/ReaderPostServiceRemote+CardsTests.swift

```diff
@@ -93,4 +93,47 @@ class ReaderPostServiceRemoteCardTests: RemoteTestCase, RESTTestable {
 
         XCTAssertTrue(mockRemoteApi.URLStringPassedIn?.contains("&page_handle=foobar") ?? false)
     }
+    
+    // Calls the API with the given sorting option
+    //
+    func testCallAPIWithTheGivenSortingOption() {
+        let readerPostServiceRemote = ReaderPostServiceRemote(wordPressComRestApi: mockRemoteApi)
+
+        readerPostServiceRemote.fetchCards(for: [], sortingOption: "foobar", success: { _, _ in }, failure: { _ in })
+
+        XCTAssertTrue(mockRemoteApi.URLStringPassedIn?.contains("sort=foobar") ?? false)
+    }
+    
+    // Calls the API without the given sorting option
+    //
+    func testCallAPIWithoutTheGivenSortingOption() {
+        let readerPostServiceRemote = ReaderPostServiceRemote(wordPressComRestApi: mockRemoteApi)
+
+        readerPostServiceRemote.fetchCards(for: [], sortingOption: nil, success: { _, _ in }, failure: { _ in })
+        
+        XCTAssertFalse(mockRemoteApi.URLStringPassedIn?.contains("sort=") ?? false)
```

Comment:
is this condition enough to guarantee that the test actually succeeds? What happens if `mockRemoteApi.URLStringPassedIn` is `nil`? Should we actually check that the `GET` method is called (maybe in other tests too)?

Reply:
If the `mockRemoteApi.URLStringPassedIn` is nil the test will fail.
By checking if the `GET` method is called you mean to fail the test if the `failure` block of the `fetchCards` is called, check the `getMethodCalled` (`XCTAssertTrue(mockRemoteApi.getMethodCalled)`) or some other way?

Reply:
> If the mockRemoteApi.URLStringPassedIn is nil the test will fail.

it seems to me that if `mockRemoteApi.URLStringPassedIn` is `nil`, then `mockRemoteApi.URLStringPassedIn?.contains("sort=") ?? false` would fall back to `false` and then `XCTAssertFalse(mockRemoteApi.URLStringPassedIn?.contains("sort=") ?? false)` would succeed, right?

Reply:
> By checking if the GET method

I believe the Mock API has a boolean to check that the method has actually been called (although an `expectation` would probably be preferable, but we can use that boolean too)

Reply:
> > If the mockRemoteApi.URLStringPassedIn is nil the test will fail.
> 
> it seems to me that if `mockRemoteApi.URLStringPassedIn` is `nil`, then `mockRemoteApi.URLStringPassedIn?.contains("sort=") ?? false` would fall back to `false` and then `XCTAssertFalse(mockRemoteApi.URLStringPassedIn?.contains("sort=") ?? false)` would succeed, right?

Oh, now I understand. Since I copy/pasted those `XCTAssertTrue` cases from other tests I forgot to change the fallback bool value. Thank you for catching it!